presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -319,6 +342,14 @@ void PrestoServer::run() {
           proxygen::ResponseHandler* downstream) {
         server->reportServerInfo(downstream);
       });
+  httpServer_->registerGet(
+      "/v1/info/health/metrics",
+      [server = this](
+          proxygen::HTTPMessage* /*message*/,
+          const std::vector<std::unique_ptr<folly::IOBuf>>& /*body*/,
+          proxygen::ResponseHandler* downstream) {
+        server->reportHealthMetrics(downstream);
```

Comment:
Why do we need this? If we just define an [exposer](https://github.com/jupp0r/prometheus-cpp/blob/master/pull/src/exposer.cc) that will create another http server on /metrics or /v1/metrics? We can also define the number of threads to be used by the new http server

Reply:
That is exactly what we are trying to avoid. We didn't find a strong need to start a new server sharing space with our worker containers. Reduces the overhead of maintaining and launching another process. Why do you want multiple threads to be spawned by the exposer? are you expecting high traffic.
CMIW, prometheus server is configured to periodically call the scrape endpoint. So, we can expect 1 HTTP request at X seconds interval. If you are expecting huge traffic, then it is not recommended to share space with worker instance.

Reply:
I think isolation is also one point, we do not want to degrade the presto process as much as we can

Reply:
@amitkdutta : What approach did Meta take for this ? Does your metric collection use any endpoint in Presto worker process itself, or did you start another server ?